axelor-sale/src/main/java/com/axelor/apps/sale/service/loyalty/LoyaltyAccountServiceImpl.java

```diff
@@ -0,0 +1,21 @@
+package com.axelor.apps.sale.service.loyalty;
+
+import com.axelor.apps.base.db.Company;
+import com.axelor.apps.base.db.Partner;
+import com.axelor.apps.sale.db.LoyaltyAccount;
+
+public class LoyaltyAccountServiceImpl implements LoyaltyAccountService {
+
+  @Override
+  public LoyaltyAccount getLoyaltyAccount(Partner partner, Company company) {
```

Comment:
You may directly return the optional here because it can be null in the end, it will be better to prevent NPE and manage empty value in methods that call `getLoyaltyAccount`